File: app/src/main/java/com/cere/csplayer/view/RecyclerViewDivider.java
package com.cere.csplayer.view;

import android.graphics.Canvas;
import android.graphics.Rect;
import android.graphics.drawable.ColorDrawable;
import android.graphics.drawable.Drawable;
import android.view.View;

import androidx.annotation.ColorInt;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.simplecityapps.recyclerview_fastscroll.views.FastScrollRecyclerView;

/**
 * Created by CheRevir on 2019/10/24
 */
public class RecyclerViewDivider extends FastScrollRecyclerView.ItemDecoration {
    private Drawable mDivider;
    private int mDividerWeight = 4;

    public RecyclerViewDivider(@ColorInt int dividerColor) {
        this(dividerColor, 4);
    }

    public RecyclerViewDivider(@ColorInt int dividerColor, int dividerWeight) {
        mDivider = new ColorDrawable(dividerColor);
        mDividerWeight = dividerWeight;
    }

    @Override
    public void onDraw(@NonNull Canvas c, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        super.onDraw(c, parent, state);
        final int left = parent.getPaddingLeft();
        final int right = parent.getWidth() - parent.getPaddingRight();

        final int childCount = parent.getChildCount();
        for (int i = 0; i < childCount; i++) {
            final View child = parent.getChildAt(i);
            final RecyclerView.LayoutParams params = (RecyclerView.LayoutParams) child.getLayoutParams();
            final int top = child.getBottom() + params.bottomMargin;
            final int bottom = top + mDivider.getIntrinsicHeight() + mDividerWeight;
            mDivider.setBounds(left, top, right, bottom);
            mDivider.draw(c);
        }
    }

    @Override
    public void getItemOffsets(@NonNull Rect outRect, @NonNull View view, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        outRect.set(0, 0, 0, mDivider.getIntrinsicHeight() + mDividerWeight);
    }
}

File: app/src/main/java/com/cere/csplayer/view/RecyclerViewDivider.kt
package com.cere.csplayer.view

import android.graphics.Canvas
import android.graphics.Rect
import android.graphics.drawable.ColorDrawable
import android.graphics.drawable.Drawable
import android.view.View
import androidx.annotation.ColorInt
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ItemDecoration

/**
 * Created by CheRevir on 2019/10/24
 */
class RecyclerViewDivider constructor(@ColorInt dividerColor: Int, dividerWeight: Int = 4) : ItemDecoration() {
    private val mDivider: Drawable
    private val mDividerWeight: Int

    init {
        mDivider = ColorDrawable(dividerColor)
        mDividerWeight = dividerWeight
    }

    override fun onDraw(c: Canvas, parent: RecyclerView, state: RecyclerView.State) {
        super.onDraw(c, parent, state)
        val left = parent.paddingLeft
        val right = parent.width - parent.paddingRight
        val childCount = parent.childCount
        for (i in 0 until childCount) {
            val child = parent.getChildAt(i)
            val params = child.layoutParams as RecyclerView.LayoutParams
            val top = child.bottom + params.bottomMargin
            val bottom = top + mDivider.intrinsicHeight + mDividerWeight
            mDivider.setBounds(left, top, right, bottom)
            mDivider.draw(c)
        }
    }

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        outRect[0, 0, 0] = mDivider.intrinsicHeight + mDividerWeight
    }
}